Extension/src/githubAPI.ts

```
@@ -177,6 +177,54 @@ function getTargetBuild(builds: Build[], updateChannel: string): Build {
     return needsUpdate(userVersion, targetVersion) ? targetBuild : undefined;
 }
 
+interface Rate {
+    remaining: number;
+}
+
+interface RateLimit {
+    rate: Rate;
+}
+
+function isRate(input: any): input is Rate {
+    return input && input.remaining && typeof(input.remaining) === 'number';
+}
+
+function isRateLimit(input: any): input is RateLimit {
+    return input && input.rate && isRate(input.rate);
+}
+
+async function getRateLimit(): Promise<RateLimit> {
+    try {
+        const header: OutgoingHttpHeaders = { 'User-Agent': 'vscode-cpptools' };
+        const data: string = await util.downloadFileToStr('https://api.github.com/rate_limit', header)
+            .catch(() => { throw new Error('Failed to download rate limit JSON'); });
+
+        let rateLimit: any;
+        try {
+            rateLimit = JSON.parse(data);
+        } catch (error) {
+            throw new Error('Failed to parse rate limit JSON');
+        }
+
+        if (isRateLimit(rateLimit)) {
+            return Promise.resolve(rateLimit);
+        } else {
+            throw new Error('Rate limit JSON is not of type RateLimit');
+        }
+    } catch (error) {
```

Since this is an `async` function using `await`, you shouldn't need to catch/return reject explicitly here.  Just let the exception be thrown out of this method.  (e.g. remove the outer try/catch)